src/Lean/Elab/BuiltinCommand.lean

```diff
@@ -118,6 +118,74 @@ private partial def elabChoiceAux (cmds : Array Syntax) (i : Nat) : CommandElabM
 @[builtin_command_elab choice] def elabChoice : CommandElab := fun stx =>
   elabChoiceAux stx.getArgs 0
 
+/-- Declares one or more universe variables.
+
+`Prop`, `Type`, `Type u` and `Sort u` are types that classify other types, also known as
+*universes*. In `Type u` and `Sort u`, the variable `u` stands for the universe's *level*, and a
+universe at level `u` can only classify universes that are at levels lower than `u`. For more
+details on type universes, please refer to [the relevant chapter of Theorem Proving in Lean][tpil
+universes].
+
+Just as type arguments allow polymorphic definitions to be used at many different types, universe
+parameters, represented by universe variables, allow a definition to be used at any required level.
+While Lean mostly handles universe levels automatically, declaring them explicitly can provide more
+control when writing signatures. The universe keyword allows the declared universe variables to be
+used in a collection of definitions, and Lean will ensure that these definitions use them
+consistently.
+
+[tpil universes]: https://lean-lang.org/theorem_proving_in_lean4/dependent_type_theory.html#types-as-objects
+(Type universes on Theorem Proving in Lean)
+
+```lean
+/- Explicit type-universe parameter. -/
+def id₁.{u} (α : Type u) (a : α) := a
+
+/- Implicit type-universe parameter, equivalent to `id₁`.
+  Requires option `autoImplicit true`, which is the default. -/
+def id₂ (α : Type u) (a : α) := a
+
+/- Explicit standalone universe variable declaration, equivalent to `id₁` and `id₂`. -/
+universe u
+def id₃ (α : Type u) (a : α) := a
+
+/- A more involved example: `WitnessPair` stores a `Sort v` and a `Sort v'`, meaning it must be in a
+  universe above both `v` and `v'`, meaning it must be at least at level `(max v v') + 1`.
+
+  So `WitnessPair` is (at least) in `Sort ((max v v') + 1)`, which is the same as `Type (max v v')`.
+-/
+universe v v'
+structure WitnessPair : Type (max v v') where
+  α : Sort v
+  val : α
+  α' : Sort v'
+  val' : α'
```

Comment:
I think this would be better down the bottom in the `Examples` section; or just removed entirely; this is not really about the `universe` keyword at all at this  point, as `structure WitnessPair.{v,v'}` would demonstrate the same thing.

Reply:
Right, it attempts to give a more interesting universe example for newbies, but I agree that ideally this should be in a manual somewhere with a broader discussion of universes.

But the thing is, we already reference TPIL which does provide a discussion of universes, so I think we might as well just drop it.

Reply:
Fixed